src/pyFAI/detectors/_dectris.py

```diff
@@ -97,11 +106,32 @@ def get_config(self):
 
         :return: dict with param for serialization
         """
-        dico = {"orientation": self.orientation}
-        if self.sensor:
-            dico["sensor"] = self.sensor.as_dict()
+        dico = super().get_config()
+
+        if ((self.module_size is not None) and
+                (tuple(self.module_size) != tuple(self.__class__.MODULE_SIZE))):
+            dico["module_size"] = self.module_size
         return dico
 
+    def set_config(self, config:dict|str):
+        """set the config of the detector
+
+        For Dectris detector, possible keys are: max_shape, module_size, orientation, sensor
+
+        :param config: dict or JSON serialized dict
+        :return: Eiger instance
+        """
+        config = _ensure_dict(config)
+        # pixel size is enforced by the detector itself
+        if "max_shape" in config:
+            self.max_shape = tuple(config["max_shape"])
+        module_size = config.get("module_size")
+        if module_size is not None:
+            self.module_size = tuple(module_size)
+        self._orientation = Orientation(config.get("orientation", 3))
+        self.sensor = SensorConfig(config["sensor"]) if config.get("sensor") is not None else None
+        return self
+
     def __repr__(self):
```

Comment:
```py
    def __repr__(self):
        """Nice representation of the instance
        """
        txt = f"Detector {self.name}"
        if self.splinefile:
            txt += f"\t Spline= {self.splinefile}"
        if (self._pixel1 is None) or (self._pixel2 is None):
            return "Undefined detector"
        else:
            txt += f"\t PixelSize= {to_eng(self._pixel1)}m, {to_eng(self._pixel2)}m"
        if self.orientation:
            txt += f"\t {self.orientation.name} ({self.orientation.value})"
        if self.sensor:
            txt += f"\t {self.sensor}"
        return txt
```
This method is from the `Detector` class. Would `_Dectris` need its own `__repr__` method?